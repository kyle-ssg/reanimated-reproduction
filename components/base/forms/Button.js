import React, { PureComponent } from 'react';
import propTypes from 'prop-types';

import cn from 'classnames';

const _propTypes = {
  /** What class to add to the button */
  className: propTypes.string,
  /** The element's children */
  children: propTypes.node,
  /** What to do on click */
  onClick: propTypes.func,
};

/**
 * Default Button without any styles
 */
export class Button extends PureComponent {
  static displayName = 'Button';

  static propTypes = _propTypes

  render() {
    const { children, ...rest } = this.props;
    return (
        <button
          type="button"
          {...rest}
          onMouseUp={this.onMouseUp}
          className={cn({
              'btn': true,
            }, (this.props.className))}
        >
            {children}
        </button>
    );
  }
}
global.Button = Button;
export default Button;

/** Default button added btn-primary * */
export class ButtonPrimary extends React.PureComponent {
  static displayName = 'ButtonPrimary';

  static propTypes = _propTypes;

  render() {
    const { props } = this;
    return (
        <Button
          {...props}
          className={cn(props.className, 'btn btn-primary')}
        />
    );
  }
}
global.ButtonPrimary = ButtonPrimary;

/** Default button added btn-secondary * */
export class ButtonSecondary extends React.PureComponent {
  static displayName = 'ButtonSecondary';

  static propTypes = _propTypes;

  render() {
    const { props } = this;
    return (
        <Button
          {...props}
          className={cn(props.className, 'btn btn-secondary')}
        />
    );
  }
}
global.ButtonSecondary = ButtonSecondary;

/** Default button added btn-outline-primary * */
export class ButtonTertiary extends React.PureComponent {
  static displayName = 'ButtonButtonTertiary';

  static propTypes = _propTypes;

  render() {
    const { props } = this;
    return (
        <Button
          {...props}
          className={cn(props.className, 'btn btn-outline-primary')}
        />
    );
  }
}

global.ButtonTertiary = ButtonTertiary;
