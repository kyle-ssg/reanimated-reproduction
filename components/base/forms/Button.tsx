import React from "react";
import cn from "classnames";

interface Button {
  className?: string,
  children?: React.ReactChildren | React.ReactChild,
  onClick?: () => void,
  onMouseUp?: () => void,
  type?: "button" | "submit" | "reset";
  disabled?: boolean;
}

//Default Button without any styles
export const Button: React.FC<Button> = ({ className, children, onMouseUp, ...rest }) => (
    <button 
      type="button" {...rest} 
      onMouseUp={onMouseUp} 
      className={cn({ btn: true }, className)}
    >
        {children}
    </button>
);

// I'm not sure what should do with this.
// global.Button = Button;
Button.displayName = "Button"
export default Button;


/** Default button added btn-primary * */
export const ButtonPrimary: React.FC<Button> = ({ className, ...props }) => <Button {...props} className={cn(className, "btn btn-primary")} />

// I'm not sure what should do with this.
// global.ButtonPrimary = ButtonPrimary;
ButtonPrimary.displayName = "ButtonPrimary";


/** Default button added btn-secondary * */
export const ButtonSecondary: React.FC<Button> = ({ className ,...props }) => <Button {...props} className={cn(className, "btn btn-secondary")} />

// I'm not sure what should do with this.
// global.ButtonSecondary = ButtonSecondary;
ButtonSecondary.displayName = "ButtonSecondary";


/** Default button added btn-outline-primary * */
export const ButtonTertiary: React.FC<Button> = ({ className, ...props }) => <Button {...props} className={cn(className, "btn btn-outline-primary")}/>

// I'm not sure what should do with this.
// global.ButtonTertiary = ButtonTertiary;
ButtonTertiary.displayName = "ButtonButtonTertiary";

