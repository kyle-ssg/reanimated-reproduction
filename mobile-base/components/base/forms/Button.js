'use strict';

var Button = Component({
    displayName: 'Button',
    propTypes: {
        onPress: OptionalFunc, //What to do on press
        onPressIn: OptionalFunc, //What to do on press in
        onPressOut: OptionalFunc, //What to do on press out
        onLongPress: OptionalFunc, //What to do on long press
        style: OptionalObject, // style for the button container
        textStyle: OptionalObject, // style for the button text
        disabled: OptionalBool, // whether the button is disabled
        variation: OptionalString // a way to use predefined style variations (e.g. large, warning)
    },
    render: function() {
        var touchableProps = {
            activeOpacity: this._computeActiveOpacity(),
        };

        if (!this.props.disabled) {
            touchableProps.onPress = this.props.onPress || this.props.onClick;
            touchableProps.onPressIn = this.props.onPressIn;
            touchableProps.onPressOut = this.props.onPressOut;
            touchableProps.onLongPress = this.props.onLongPress;
        }


        //compute styles e.g. buttonGroupLeft, big, bigRight, buttonGroupText, bigText, but
        var groupStyle = [styles.buttonGroup,
            this.props.position && styles['buttonGroup' + Format.camelCase(this.props.position)],
            this.props.variation && styles['buttonGroup' + Format.camelCase(this.props.variation)],
            this.props.variation && this.props.position && styles[this.props.variation + Format.camelCase(this.props.position)],
            this.props.style
        ];

        var textStyle = [styles.buttonText,
            this.props.position && styles['buttonText' + Format.camelCase(this.props.position) + "Text"],
            this.props.variation && styles['buttonText' + Format.camelCase(this.props.variation)],
            this.props.variation && this.props.position && styles[this.props.variation + Format.camelCase(this.props.position) + "Text"],
            this.props.textStyle
        ];


        return (
            <View style={{opacity: this.props.disabled? 0.5: 1}}>
                <TouchableOpacity {...touchableProps}
                    style={groupStyle}>
                    { typeof this.props.children == "string" ? (
                        <Text style={textStyle}>{this.props.children}</Text>
                    ) : this.props.children}
                </TouchableOpacity>
            </View>
        );
    },

    _computeActiveOpacity() {
        if (this.props.disabled) {
            return 1;
        }
        return colour.buttonActiveOpacity;
    },
});

var styles = require('../../../style/style_buttons')

module.exports = Button;