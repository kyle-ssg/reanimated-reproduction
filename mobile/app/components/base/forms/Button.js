import React, { PureComponent } from 'react';
import propTypes from 'prop-types';
import { StyleSheet, TouchableNativeFeedback } from 'react-native';

const buttonPropTypes = {
    accessibilityLabel: propTypes.string, // Accessibility label
    accessible: propTypes.bool, // Whether the button is accessible
    onPress: propTypes.func, // What to do on press
    onClick: propTypes.func,
    onPressIn: propTypes.func, // What to do on press in
    onPressOut: propTypes.func, // What to do on press out
    onLongPress: propTypes.func, // What to do on long press
    style: propTypes.any,
    textStyle: propTypes.any, // style for the button text
    disabled: propTypes.bool, // whether the button is disabled
    variation: propTypes.string, // a way to use predefined style variations (e.g. large, warning)
    testID: propTypes.string, // used for e2e testing
    children: propTypes.oneOfType([
        propTypes.arrayOf(propTypes.node),
        propTypes.node,
    ]).isRequired,
};

export default class Button extends PureComponent {
  static displayName = 'Button';

  static propTypes = buttonPropTypes;

  _computeActiveOpacity() {
      if (this.props.disabled) {
          return 1;
      }
      return colour.buttonActiveOpacity;
  }

  render() {
      const touchableProps = {
          activeOpacity: this._computeActiveOpacity(),
      };

      if (!this.props.disabled) {
          touchableProps.onPress = this.props.onPress || this.props.onClick;
          touchableProps.onPressIn = this.props.onPressIn;
          touchableProps.onPressOut = this.props.onPressOut;
          touchableProps.onLongPress = this.props.onLongPress;
      }

      // compute styles e.g. buttonGroupLeft, big, bigRight, buttonGroupText, bigText, but
      const groupStyle = [Styles.buttonGroup, this.props.style];

      const baseTextStyle = { color: 'white', letterSpacing: 2, fontSize: 14 };
      const additionalTextStyles = this.props.textStyle && this.props.textStyle.length ? this.props.textStyle : [this.props.textStyle];

      const textStyle = [
          baseTextStyle,
          ...additionalTextStyles,
      ];

      return Platform.OS === 'android' && Platform.Version >= 21 ? (
          <View style={{ opacity: this.props.disabled ? 0.5 : 1 }}>
              <TouchableNativeFeedback
                {...touchableProps}
                background={TouchableNativeFeedback.Ripple('rgba(255,255,255,.5)')}
                testID={this.props.testID}
                accessible={this.props.accessible}
                accessibilityLabel={this.props.accessibilityLabel}
              >
                  <View style={groupStyle}>
                      {typeof this.props.children === 'string' ? (
                          <Text style={textStyle}>{this.props.children.toUpperCase()}</Text>
                      ) : (
                          this.props.children
                      )}
                  </View>
              </TouchableNativeFeedback>
          </View>
      ) : (
          <View style={{ opacity: this.props.disabled ? 0.5 : 1 }}>
              <TouchableOpacity
                {...touchableProps}
                style={[
                    groupStyle,
                    this.props.disabled && { backgroundColor: pallette.primaryDarkAlt },
                ]}
                accessible={this.props.accessible}
                accessibilityLabel={this.props.accessibilityLabel}
                testID={this.props.testID}
              >
                  {typeof this.props.children === 'string' ? (
                      <Text pointerEvents="none" style={textStyle}>
                          {this.props.children}
                      </Text>
                  ) : (
                      this.props.children
                  )}
              </TouchableOpacity>
          </View>
      );
  }
}

export const ButtonPrimary = (props) => (
    <Button
      {...props}
      style={[styles.buttonPrimary, props.style]}
    />
);
ButtonPrimary.propTypes = buttonPropTypes;

export const ButtonPrimaryBookmaker = (props) => (
    <Button
      {...props}
      style={[{ backgroundColor: pallette.bookmakerPrimary }, props.style]}
    />
);
ButtonPrimaryBookmaker.propTypes = buttonPropTypes;

export const ButtonSecondary = (props) => (
    <Button
      {...props}
      style={[{ backgroundColor: pallette.secondary }, props.style]}
    />
);
ButtonSecondary.propTypes = buttonPropTypes;

export const ButtonTertiary = (props) => (
    <Button
      {...props}
      style={[styles.buttonTertiary,
          props.style,
      ]}
      textStyle={[styles.buttonTertiaryText, props.textStyle]}
    />
);
ButtonTertiary.propTypes = buttonPropTypes;

export const ButtonTertiaryBookmaker = (props) => (
    <Button
      {...props}
      style={[
          styles.buttonTertiaryBookmaker,
          props.style,
      ]}
      textStyle={[styles.buttonTertiaryBookmakerText, props.textStyle]}
    />
);
ButtonTertiaryBookmaker.propTypes = buttonPropTypes;

export const ButtonAlt = (props) => (
    <Button
      {...props}
      style={[{ backgroundColor: pallette.primaryDark }, props.style]}
    />
);
ButtonAlt.propTypes = buttonPropTypes;

export const ButtonFlat = (props) => (
    <Button {...props} style={[Styles.buttonWhiteShadow, props.style]} />
);
ButtonFlat.propTypes = buttonPropTypes;

export const ButtonDashed = (props) => (
    <Button {...props} style={[Styles.buttonDashed, props.style]} />
);
ButtonDashed.propTypes = buttonPropTypes;

export const ButtonOutlinePrimary = (props) => (
    <Button {...props} style={[Styles.buttonOutlinePrimary, props.style]} />
);
ButtonOutlinePrimary.propTypes = buttonPropTypes;

export const ButtonGoogle = (props) => (
    <Button
      {...props}
      style={[styles.buttonGoogle, props.style]}
    />
);
ButtonGoogle.propTypes = buttonPropTypes;

export const ButtonCommon = (props) => (
    <Button
      {...props}
      style={[styles.buttonCommon, props.style]}
      textStyle={[{
          color: pallette.text,
      }, props.textStyle]}
    />
);
ButtonCommon.propTypes = buttonPropTypes;

export const TabPillButton = (props) => (
    <Button {...props} style={[styles.TabButtonPill, props.style]}>
        <Text
          pointerEvents="none"
          style={[styles.TabButtonPillText, props.textStyle]}
        >
            {props.children}
        </Text>
    </Button>
);
TabPillButton.propTypes = buttonPropTypes;

export const LinkTextButton = (props) => (
    <Button {...props} style={[styles.LinkTextButton, props.style]}>
        <Text
          pointerEvents="none"
          style={[styles.LinkTextButtonText, props.textStyle]}
        >
            {props.children}
        </Text>
    </Button>
);
LinkTextButton.propTypes = buttonPropTypes;

const styles = StyleSheet.create({
    TabButtonPill: {
        height: 34,
        flexWrap: 'wrap',
        alignItems: 'center',
        flexDirection: 'row',
    },
    TabButtonPillText: {
        color: pallette.primaryDark,
    },
    LinkTextButton: {
        backgroundColor: 'transparent',
        height: 'auto',
    },
    buttonTertiary: {
        backgroundColor: 'white',
        borderColor: pallette.primary,
        borderWidth: 1,
    },
    buttonTertiaryText: {
        color: pallette.primary,
    },
    buttonTertiaryBookmaker: {
        backgroundColor: 'white',
        borderColor: pallette.bookmakerPrimaryDark,
        borderWidth: 1,
    },
    buttonCommon: {
        backgroundColor: pallette.racingPostLightGray,
    },
    buttonTertiaryBookmakerText: {
        color: pallette.bookmakerPrimaryDark,
    },
    buttonGoogle: { backgroundColor: pallette.primaryDanger },
    LinkTextButtonText: {
        color: pallette.linkBlue,
    },
    buttonPrimary: { backgroundColor: pallette.primary },
});
