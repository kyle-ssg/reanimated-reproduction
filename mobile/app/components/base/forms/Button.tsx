import {
  Platform,
  Pressable,
  RippleBackgroundPropType,
  StyleSheet,
  TouchableNativeFeedback,
  TouchableOpacity
} from "react-native";

import Utils from "common/utils/base/_utils";
import throttle from "lodash/debounce";
import React, { FunctionComponent, useMemo } from "react";
import useTheme from "common/providers/useTheme";

export type ButtonType = ReactNative.PressableProps & {
  children: React.ReactNode,
  icon?: boolean,
  textStyle?: ReactNative.TextStyle | ReactNative.TextStyle[],
  pressedStyle?: ReactNative.ViewStyle | ReactNative.ViewStyle[],
  containerStyle?: ReactNative.ViewStyle | ReactNative.ViewStyle[],
  pressedTextStyle?: ReactNative.TextStyle | ReactNative.TextStyle[],
  style?: ReactNative.ViewStyle
  throttle?: number
};

export const standardAndroidRipple: ReactNative.PressableAndroidRippleConfig = {
  color: "rgba(255,255,255,.25)",
  borderless: false
};

const circleButtonRipple: ReactNative.PressableAndroidRippleConfig = {
  color: "rgba(0,0,0,.15)",
  borderless: true
};

export const darkAndroidRipple: ReactNative.PressableAndroidRippleConfig = {
  color: "rgba(0,0,0,.05)",
  borderless: false
};

const Button: FunctionComponent<ButtonType> = ({
  disabled,
  android_ripple,
  children,
  style,
  pressedStyle,
  pressedTextStyle,
  containerStyle,
  textStyle,
  icon,
  onPress,
  throttle: _throttle = 500,
  ...rest
}) => {
  const groupStyles = useMemo(() => (
    style ? [Styles.buttonGroup, style, disabled ? Styles.buttonDisabled : null] : [Styles.buttonGroup, disabled ? Styles.buttonDisabled : null]
  ), [style, disabled]);

  const onPressThrottle = onPress && throttle(onPress, _throttle, { leading: true });

  const pressedStyles = useMemo(() => (
    [Styles.buttonGroup, Styles.buttonGroupPressed, style, pressedStyle]
  ), [pressedStyle, style]);

  const textStyles = useMemo(() => {
    // @ts-ignore
    const additionalTextStyles = textStyle && textStyle?.length ? textStyle : [textStyle];
    return textStyle ? [
      Styles.buttonText,
      // @ts-ignore
      ...additionalTextStyles
    ] : Styles.buttonText;
  }, [textStyle]);


  return (
    <View style={[{ overflow: "hidden", position: "relative" }, containerStyle]}>
      <Pressable
        {...rest}
        onPress={onPressThrottle}
        style={({ pressed }) => pressed ? pressedStyles : groupStyles}
        disabled={disabled}
        android_ripple={android_ripple || darkAndroidRipple}
      >
        {Utils.reactChildIsString(children) ?
          ({ pressed }) => (
            // @ts-ignore
            <Text style={!pressed || !pressedTextStyle ? textStyles : [...textStyles, pressedTextStyle]}>
              {/*@ts-ignore*/}
              {children.length === 1 ? children[0] : children}
            </Text>
          )
          : (children)

        }
      </Pressable>
      {icon ?
        <View style={{
          position: "absolute",
          right: 10,
          width: 30,
          height: 30,
          top: 10,
          borderRadius: styleVariables.baseBorderRadius,
          justifyContent: "center",
          alignItems: "center"
        }}
        >
          <FA5Pro solid name={icon} size={18}
            color={palette.white}
          />
        </View>
        :
        null
      }
    </View>
  );
};


export const ButtonPrimary: FunctionComponent<ButtonType> = (props) => {
  return (
    <Button
      {...props}
      style={[Styles.buttonPrimary, props.style]}
      // @ts-ignore
      pressedStyle={[Styles.buttonPrimaryPressed, props.pressedStyle]}
      // @ts-ignore
      textStyle={[Styles.buttonPrimaryText, props.textStyle]}
    />
  );
};

export const ButtonOutlinePrimary: FunctionComponent<ButtonType> = (props) => {
  return (
    <Button
      {...props}
      style={[Styles.buttonOutlinePrimary, props.style]}
      // @ts-ignore
      pressedStyle={[Styles.buttonOutlinePrimaryPressed, props.pressedStyle]}
      // @ts-ignore
      textStyle={[Styles.buttonOutlinePrimaryText, props.textStyle]}
      pressedTextStyle={Styles.buttonOutlinePrimaryPressedText}
    />
  );
};

export const ButtonLink: FunctionComponent<ButtonType> = (props) => {
  return (
    <Button
      {...props}
      android_ripple={{ color: "transparent" }}
      style={[Styles.buttonLink, props.style]}
      // @ts-ignore
      pressedStyle={[Styles.buttonLinkPressed, props.pressedStyle]}
      // @ts-ignore
      textStyle={[Styles.buttonLinkText, props.textStyle]}
      pressedTextStyle={Styles.buttonLinkPressedText}
    />
  );
};

export const ButtonSecondary: FunctionComponent<ButtonType> = (props) => {
  return (
    <Button
      {...props}
      style={[Styles.buttonSecondary, props.style]}
      // @ts-ignore
      pressedStyle={[Styles.buttonSecondaryPressed, props.pressedStyle]}
    />
  );
};

export const ButtonTertiary: FunctionComponent<ButtonType> = (props) => {
  return (
    <Button
      {...props}
      style={[Styles.buttonTertiary, props.style]}
      // @ts-ignore
      pressedStyle={[Styles.buttonTertiaryPressed, props.pressedStyle]}
      // @ts-ignore
      textStyle={[Styles.buttonTertiaryText, props.textStyle]}
    />
  );
};

export const ButtonText: FunctionComponent<ButtonType> = (props) => {

  return (
    <Button
      {...props}
      style={[Styles.buttonText, props.style]}
      // @ts-ignore
      textStyle={[Styles.buttonTextText, props.textStyle]}
    />
  );
};

export const ButtonNav: FunctionComponent<ButtonType> = (props) => {
  return (
    <Button
      {...props}
      android_ripple={circleButtonRipple}
      style={[props.style, Styles.buttonText]}
    />
  );
};

const styles = ReactNative.StyleSheet.create({});

export default Button;
