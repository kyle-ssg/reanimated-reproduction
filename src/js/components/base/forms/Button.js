const TheComponent = class extends React.Component {
    displayName:'TheComponent'

    constructor (props, context) {
        super(props, context);
        this.state = {};
    }

    onMouseUp =() => {
        this.refs.button.blur();
    }

    render () {
        return (
            <button ref="button" {... this.props} onMouseUp={this.onMouseUp} className={"btn  " + (this.props.className || "btn-primary")}>
                {this.props.children}
            </button>
        );
    }
};

TheComponent.propTypes = {};

module.exports = TheComponent;