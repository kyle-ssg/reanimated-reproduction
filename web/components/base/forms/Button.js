import React from 'react';
import cn from 'classnames';

global.Button = class extends React.Component {
    static displayName = 'Button';

    constructor(props, context) {
        super(props, context);
        this.state = {};
    }

    onMouseUp = () => {
        this.refs.button.blur();
    };

    render() {
        const { children, ...rest } = this.props;
        return (
            <button
              ref="button"
              type="button"
              {...rest}
              onMouseUp={this.onMouseUp}
              className={cn({
                  'btn': true,
              }, (this.props.className))}
            >
                {children}
            </button>
        );
    }
}

export default Button;

export const ButtonPrimary = global.ButtonPrimary = props => (
    <Button
      {...props}
      className={cn(props.className, 'btn btn-primary')}
    >
        {props.children}
    </Button>
);

export const ButtonSecondary = global.ButtonSecondary = props => (
    <Button
      {...props}
      className={cn(props.className, 'btn btn-secondary')}
    >
        {props.children}
    </Button>
);

export const ButtonTertiary = global.ButtonTertiary = props => (
    <Button
      {...props}
      className={cn(props.className, 'btn btn-outline-primary')}
    >
        {props.children}
    </Button>
);
