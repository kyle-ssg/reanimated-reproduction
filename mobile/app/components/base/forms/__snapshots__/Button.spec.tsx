import * as React from 'react'
import Button from '../Button'
import renderer from 'react-test-renderer'

describe('<Button />', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<Button />)
    expect(tree).toMatchSnapshot()
  })
})
