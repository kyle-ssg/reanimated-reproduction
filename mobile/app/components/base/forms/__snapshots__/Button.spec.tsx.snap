// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#1AC0C6",
          "borderRadius": undefined,
          "height": 48,
          "justifyContent": "center",
          "overflow": "hidden",
        },
        null,
      ]
    }
  />
</View>
`;
